### ISSUE TITLE
Accept empty subcommunities

### ISSUE THREAD

**Issue by IosiaLectus** [ArnaoutLab/diversity#106]
I am currently getting the error

"greylock.exceptions.InvalidArgumentError: Argument 'weights' must have at least one nonzero weight in each column. A weight is considered 0 if its absolute value is greater than or equal to configurable minimum threshold: 1.00e-09."

I would prefer if this were a warning, and greylock would just give zeros for the relevant measures.

**Comment by chhotii-alex** [comment]
Yes, agreed

**Comment by chhotii-alex** [comment]
For q = 0 the calculations as currently implemented do the right thing (return 0.0). However for other q there's division by zero errors and numerical badness and `inf` is given as the result much of the time.

I'm thinking, just add `np.finfo(np.float64).tiny` to all the counts just before the relevant calculations. For any number that's larger than a negligible size, this is a no-op (the number is unchanged, because the interval between representable floats gets bigger). If the count is zero, calculations on tiny wind up giving zero. Likewise if something is _so_ small that adding `tiny` changes it.

Hopefully this wouldn't be considered too much of a hack.

**Comment by chhotii-alex** [comment]
Things are sufficiently complicated that just sprinkling values of `tiny` here and there doesn't give us good results (and yes, it's a terrible hack).

I'm thinking that this is a bad idea for the same reason that computers always object to division by zero. We shouldn't be jumping to a conclusion about what you want when the result is undefined.

Now, admittedly, we would be on sound mathematical ground to define metacommunity measures thusly when there's an all-zero-count subcommunity: All measures are the same as if that subcommunity were not there. And we would also be on sound mathematical ground to say that for an all-zero-count subcommunity, `normalized_alpha` is zero (also `beta` I believe.) 

However, other measures would be undefined. If someone is using other measures then depending on what they were doing with those measures (whether plotting them, finding the maximum or minimum, averaging...), there would have to be some special-case code to keep `nan`s from propagating everywhere or otherwise messing things up. 

If your pipeline might create all-zero-count subcommunities, and thus you need some special case code, it's easy enough to put the special case code before invoking greylock:

```
counts = counts[[col for col in counts if counts[col].sum() > 0]]
```

In short, I'm now thinking that this wouldn't be a good idea.
